sdk/batch/batch/src/models/index.ts

```diff
@@ -1,14350 +1,11315 @@
+   * The registry URL. If omitted, the default is "docker.io".
    */
   registryServer?: string;
   /**
-   * @member {string} userName The user name to log into the registry server.
+   * The user name to log into the registry server.
    */
   userName: string;
   /**
-   * @member {string} password The password to log into the registry server.
+   * The password to log into the registry server.
    */
   password: string;
 }
 
 /**
- * @interface
  * An interface representing TaskContainerSettings.
  * @summary The container settings for a task.
- *
  */
 export interface TaskContainerSettings {
   /**
-   * @member {string} [containerRunOptions] Additional options to the container
-   * create command. These additional options are supplied as arguments to the
-   * "docker create" command, in addition to those controlled by the Batch
+   * Additional options to the container create command. These additional options are supplied as
+   * arguments to the "docker create" command, in addition to those controlled by the Batch
    * Service.
    */
   containerRunOptions?: string;
   /**
-   * @member {string} imageName The image to use to create the container in
-   * which the task will run. This is the full image reference, as would be
-   * specified to "docker pull". If no tag is provided as part of the image
+   * The image to use to create the container in which the task will run. This is the full image
+   * reference, as would be specified to "docker pull". If no tag is provided as part of the image
    * name, the tag ":latest" is used as a default.
    */
   imageName: string;
   /**
-   * @member {ContainerRegistry} [registry] The private registry which contains
-   * the container image. This setting can be omitted if was already provided
-   * at pool creation.
+   * The private registry which contains the container image. This setting can be omitted if was
+   * already provided at pool creation.
    */
   registry?: ContainerRegistry;
 }
 
 /**
- * @interface
  * An interface representing ResourceFile.
- * @summary A file to be downloaded from Azure blob storage to a compute node.
- *
+ * @summary A single file or multiple files to be downloaded to a compute node.
  */
 export interface ResourceFile {
   /**
-   * @member {string} blobSource The URL of the file within Azure Blob Storage.
-   * This URL must be readable using anonymous access; that is, the Batch
-   * service does not present any credentials when downloading the blob. There
-   * are two ways to get such a URL for a blob in Azure storage: include a
-   * Shared Access Signature (SAS) granting read permissions on the blob, or
-   * set the ACL for the blob or its container to allow public access.
+   * The storage container name in the auto storage account. The autoStorageContainerName,
+   * storageContainerUrl and httpUrl properties are mutually exclusive and one of them must be
+   * specified.
+   */
+  autoStorageContainerName?: string;
```

Comment:
@devardhanm - This should fix https://github.com/Azure/azure-sdk-for-js/issues/2996

Reply:
Thanks for adding this support @amarzavery